### PR TITLE
GIANT MECHS WITH SWORDS WOOOOOO

### PR DESCRIPTION
**Pull request by Mqiib**
DID YOU EVER WANT TO HAVE A GIANT SWORD ON YOUR MECH? WELL NOW YOU CAN! PACIFIC RIM THIS SHIT UP YO!
This pr adds: REALLY COOL STUFF FOR MECHS and MAKES MECH MELEE COMBAT SLIGHTLY MORE INTERESTING and also nerfs base mech melee damage a tad to make up for the fact that they'll be baller in combat now.

This is getting closer to completion
- Add mech melee weapon category
- Standard sword has a 3-tile cleave, deals +15 damage, +20 on targeted attacks, deals a minimum of 25 damage, gives passive 15 bonus deflect chance, and has 15 ap. Unlocking requires advanced melee tech and advanced exosuit tech
- Rocket fist increases mech punch damage by 20 and reduces melee cooldown by 20%. Unlocking requires basic melee tech and advanced exosuit tech.
- Energy axe shares a cleave with the sword but does more damage (70 for the mech it'll be mounted on, the mauler), has 40 AP, deals bonus damage to structures and objects, cannot be used for precise attacks, and has a 50% longer attack cooldown
- Combat mech punch damage reduced from 30 to 25 to incentivize use of melee weapons. Marauders/maulers also have had their damage reduced to 40 from 45, seraphs to 50 from 55, and durands to 35 from 40.

CREDIT TO @Moltijoe for being able to actually test this for me while my local server shits itself
TO DO:

- [X] Get the melee weapons file done
- [X] initial mech procs n stuff
- [x] Finish coding proof-of-concept sword
- [x] Swing sprite
- [x] Figure out why my test server implodes after 30 seconds
- [x] Cuackles make me a sword item sprite please
- [x] Find some _heavy_ mech sword sounds
- [x] Make sure mechs still work lol lmao
- [x] Add sword to tech web
- [x] Give maulers an axe for mauling

# Spriting
+MECH SWORD MODULES (REALLY NEAT) (THANKS CUACKLES)
image

+SLASH EFFECTS (REALLY SWOOPY)
(i dunno how to make it gif-itize but it's pretty simple)

# Wiki Documentation
**Mauler**: Now also starts with a SH-NT "Killerhurtz" energy axe and has 6 equipment slots.

**Shortsword**: Has cleave + precise attack, +15 damage on cleave, +20 on precise (minimum 25), +30 fauna damage, 2.5x structure damage multiplier, sharp, 15 base AP, and grants +15 deflection chance when the active module
**Energy axe**: has cleave, +30 damage on cleave (min 40), +30 fauna damage, 4x structure damage, sharp, 40 base AP, 1.5 cooldown multiplier, glows red, available for syndicate marauder mech
**Rocket fist**: grants +20 punch damage and 20% faster melee attacks when the active module, no special attacks

CLEAVE ATTACKS: hit the 3 front tiles of the mech (front right to front left) after a .2 second windup, targets in the area are hit in a random bodypart, solid objects take damage, damage equal to mech punch damage + base melee damage, uses base AP value, deals damage of type and has sharpness based on melee weapon module
I would get a clip of the attack but my local branch won't run for some raisin. Anyways, red tiles are the danger tiles.
image

PRECISE ATTACKS: overrides standard mech melee attack, deals base mech punch damage + precise melee damage, gains 2x AP value on the attack, deals damage of type and has sharpness based on melee weapon module

# Changelog

:cl:  Mqiib, Cuackles
rscadd: MECH SWORDS!
rscadd: Gives the mauler exosuit an energy axe for awesomeness reasons
rscadd: Adds mech melee weapons to tech web
tweak: Combat mech damage reduced to 25 from 30
tweak: Mauler/marauder, seraph, and durand mech damage reduced by 5 (45->40, 55->50, 40->35)
imageadd: BIG slash sprite
imageadd: mech melee weapon module sprites
soundadd: BIG slash noise
bugfix: fixes a dumb mecha bump issue for z-levels with nojaunt
experimental: This is experimental AND AWESOME
/:cl:

### CODE REVIEW

**Comment by githubuser4141** [comment]
Is it going to have a on-mech sprite? It's a little odd how no one save for terragov and bay has added sprited mech equipment yet

**Comment by Mqiib** [comment]
> Is it going to have a on-mech sprite? It's a little odd how no one save for terragov and bay has added sprited mech equipment yet

Maybe

**Comment by ilikesaying** [comment]
Truly a weapon to surpass Metal Gear.

**Comment by Mqiib** [comment]
Currently cant test because my local server implodes after about 30 seconds but it theoretically should work.

**Comment by Mqiib** [comment]
ok it should be actually done now

**Comment by VaelophisNyx** [comment]
Kinetic Strike Module? Pog?
zXIyWpn

**Comment by cuackles** [comment]
@KillerOrcaCora u gotta do it

File: code/game/mecha/combat/marauder.dm [hunk 1]
```diff
@@ -91,6 +91,8 @@
 	ME.attach(src)
 	ME = new /obj/item/mecha_parts/mecha_equipment/antiproj_armor_booster(src)
 	ME.attach(src)
+	ME = new /obj/item/mecha_parts/mecha_equipment/melee_weapon/sword/energy_axe(src)		//NOT BECAUSE IT'S USEFUL, BUT BECAUSE IT'S AWESOME
```
Comment (ReddicusDragon) [inline code/game/mecha/combat/marauder.dm:94]: hell yeah!

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 2]
```diff
@@ -379,8 +379,9 @@
 		apply_damage(damage, BRUTE, affecting, armor_block, wound_bonus=wound_mod)
 
 /mob/living/carbon/human/mech_melee_attack(obj/mecha/M)
-
-	if(M.occupant.a_intent == INTENT_HARM)
+	if(M.selected?.melee_override)
```
Comment (ReddicusDragon) [inline code/modules/mob/living/carbon/human/human_defense.dm:382]: What does `M.selected?` do?

**Review by ReddicusDragon — COMMENTED** [review]
Save for that one "what does `M.selected?` do", I see no issues. Willing to give a thumbs up when this is ready

File: code/modules/mob/living/carbon/human/human_defense.dm (hunk 2 above)
Comment (Mqiib) [reply code/modules/mob/living/carbon/human/human_defense.dm:382]: M.selected is the current selected module, so like if the selected module is a gun clicking will fire that gun. This checks to make sure the current module isn't a melee module that overrides the standard melee attack of the mech with something else.

**Comment by JamieD1** [comment]
This is such a bad idea - Testmerging

**Comment by Mqiib** [comment]
Ok tiny microscopic conflict resolved probably

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**